server/src/main/java/org/jboss/as/server/ElapsedTime.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright The WildFly Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.jboss.as.server;
+
+import java.lang.management.ManagementFactory;
+import java.lang.management.RuntimeMXBean;
+import java.util.Map;
+import java.util.WeakHashMap;
+
+/**
+ * Tracks elapsed time since either JVM start or a moment of initialization. The start point
+ * for the calculation can be 'reset', allowing the calculation provided by an instance to
+ * account for a conceptual 'restart'.
+ */
+public final class ElapsedTime {
+
+    /**
+     * Creates a tracker that tracks elapsed time since JVM start.
+     * @return the tracker. Will not return {@code null}.
+     */
+    public static ElapsedTime startingFromJvmStart() {
+        return new ElapsedTime(null);
+    }
+
+    /**
+     * Creates a tracker that tracks elapsed time since the invocation of this method.
+     * @return the tracker. Will not return {@code null}.
+     */
+    public static ElapsedTime startingFromNow() {
+        return new ElapsedTime(System.currentTimeMillis());
+    }
+
+    /**
+     * Creates a tracker that tracks elapsed time since whatever point the given {@code toCopy} tracker does.
+     * This is a cloning operation; the created tracker shares no state with the {@code toCopy} tracker.
+     *
+     * @param toCopy tracker to copy. Cannot be {@code null}
+     * @return the tracker. Will not return {@code null}.
+     */
+    public static ElapsedTime from(ElapsedTime toCopy) {
+        return new ElapsedTime(toCopy.startTime);
+    }
```

Comment:
I believe I can remove this now as it was for the use case that 'checkpoint()' handles now.